pkg/genlib/generator_with_custom_template.go

```diff
@@ -110,12 +167,18 @@ func (gen GeneratorWithCustomTemplate) Emit(state *GenState, buf *bytes.Buffer)
 }
 
 func (gen GeneratorWithCustomTemplate) emit(state *GenState, buf *bytes.Buffer) error {
-	for _, f := range gen.emitFuncs {
-		if err := f(state, buf); err != nil {
-			return err
+	if gen.totEvents == 0 || state.counter < gen.totEvents {
```

Comment:
@aspacca is `state.counter` updated somewhere? I've not been able to find it.

Reply:
@endorama yes, in the `Emit()`

https://github.com/elastic/elastic-integration-corpus-generator-tool/blob/c9537381950cc004c4249e30debefa819427dfba/pkg/genlib/generator_with_custom_template.go#L164
and https://github.com/elastic/elastic-integration-corpus-generator-tool/blob/c9537381950cc004c4249e30debefa819427dfba/pkg/genlib/generator_with_text_template.go#L145

it's not in the diff because nothing changed there :)